### ISSUE TITLE
Suggestions for 2017

### ISSUE THREAD

**Issue by docwilmot** [backdrop-ops/backdropcms.org#322]
Set long term goals
 - We have issue milestones for code and Backdrop CMS, good.
 - But we need to define the target audiences, what that audience would need in a full featured CMS, and collate the important things we need to do to satisfy them. Otherwise we may tend to be prioritising minor issues over the "necessary" stuff. 
 eg by September 2017 we would like 2000 active installs; our most likely market is education facilities; they would need four good themes, two good layouts, media capabilities, and SchoolMan module (made that up). Those then would become our most reasonable priorities.

Weekly meetings
 - Less frequent please. I find that the agenda seems to be a bit repetitive and gives the impression of poor progress. Most management advice speaks agains too many meetings. It would be best if the operations aspects of the conversation could happen asynchronously via some sort of discussion board (I am avoiding saying "discussion forum" because I've flogged that horse for two years now, and I've sworn to stop).
 - More structure to the agenda. I know this is voluntary, but time limits and person responsibilities would probably help push things along a bit. As it is now it tends to be the founders (re-)reading "the plan" without much clear opportunity for ownership on the part of the team.
 - Shorter meetings. because thats always good. But also because we record these and it would be nice if perople could watch them, and no one watches a 45 minute video.

Outreach
 - As noted above, I haven't listened to any of the outreach meetings in full and attending is difficult for me so maybe mentioned, but we need some paid ads.
 
The queues
 - We are adding issues at a rapid pace, and removing issues rarely. I think we should be a bit more vicious in closing issues which clearly have little chance of being done in a reasonable timeframe.
 - Some of us (me, actually) would appreciate more timely review of PRs. Hold on, don't argue yet, I know we're busy. But if we review PRs, even a brief, 'this is the right way/wrong way to do this', then us (me) less confident coders are likely to contribute a lot more. Case in point I pushed 4 PRs on November 2nd, without tests, hoping for a brief word (and I did mention this was all I was waiting for) in order to proceed; sadly, no review, and none will make it it seems. Point is with just a "change this, change that", I am reasonably certain I could have finished them on time.
 This also shows more appreciation to the community; getting my stuff 'approved' and accepted for core is a significant psychlogical reward for me at least, thats the 'pay'.
 - Less "ideas" in the queue. This would require a Forum.

**Comment by Dinornis** [comment]
> ... we need to define the target audiences, what that audience would need ...

It would be nice to know if D6 admins are on the radar as possible audience or not so they can consider other options if not. _Disclosure:_ I maintain one of the 80,000+ websites still running Drupal 6.
>  (I am avoiding saying "discussion forum" because I've flogged that horse for two years now, and I've sworn to stop).

Please, do not stop, this can't be said often enough.  An online community without a forum, seriously?
> Less "ideas" in the queue. This would require a Forum.

Exactly.

**Comment by Graham-72** [comment]
Maybe we need two or three blog/news items that are case histories of migrating sites from Drupal 6 to Backdrop? This must be the sort of thing people would be searching for and reading.

**Comment by serundeputy** [comment]
> Less "ideas" in the queue. This would require a Forum.

This one got me thinking.  I agree that it would be great to have a place to ask questions, have discussions and flesh out ideas.  I started this module: https://github.com/serundeputy/qa_forum  it is not complete and is opinionated towards a question and answer forum like stackoverflow.

> But we need to define the target audiences, what that audience would need in a full featured CMS, and collate the important things we need to do to satisfy them. Otherwise we may tend to be prioritising minor issues over the "necessary" stuff.
eg by September 2017 we would like 2000 active installs; our most likely market is education facilities; they would need four good themes, two good layouts, media capabilities, and SchoolMan module (made that up). Those then would become our most reasonable priorities.

I have also struggled with the target audience question.  I think one of the things that makes it difficult for me is that Backdrop/Drupal are designed to be flexible and meet a lot of use cases, but that flexibility comes with a complexity price tag.

To my mind our target audience is the upper tier of the Wordpress market and the lower tier of the D8 market (one persons opinion).  If you want a drop dead simple brochure/blog site I think Wordpress is a great choice, but if you have content templates (fieldable) and content relationships and need to have different layouts for different areas/contents then you are looking at Backdrop/D8 decision.  That is where my mind is at with respect to target audience, but it is of course up for debate.

**Comment by jenlampton** [comment]
> But we need to define the target audiences, what that audience would need in a full featured CMS, and collate the important things we need to do to satisfy them.

I struggle with this too, for the same reasons @serundeputy mentions. Maybe we can rotate, and choose one audience for each release? Focus one on NP, next on SM biz, next on Edu, etc? Hmm... On further thinking, this shouldn't be necessary. What all of these audiences need should be in core, what makes them different should be in contrib. 

> but we need some paid ads.

Yes, this has been on our todo list for some time. Before we start spending money on ads, we need have some way to tell if they are successful or not. What is the call to action? This gets back to having clear goals again! Maybe Facet can help us with this too?

**Comment by klonos** [comment]
> To my mind our target audience is the upper tier of the Wordpress market and the lower tier of the D8 market (one persons opinion).

I more or less feel that to be the case as well.

>Yes, a forum is a super high-priority item, #3 on our list. We need to convince people that Backdrop is real before they will talk about it in a forum, so items #1 and #2 are to that goal: The showcase, and the new home page. But those two items are itchingly-close to being ready. The forum should be item #1 soon after 1.6 is released. I'm personally very excited about it :)

👍

**Comment by Graham-72** [comment]
Here are some thoughts:

**Target markets**
Because of all the excellent UX and DX improvements that have been made, I think there are two main target markets where we can succeed.
1. The novice who wants a quick and easy way to build their own website ‘out-of-the box’.
2. The competent site-builder who values the flexibility and extendibility of Backdrop for building websites for paying clients, voluntary organisations, friends etc..  I class myself in this category.
To succeed in the first of these, I believe we need a selection of attractive ‘off-the-shelf’ themes, with very clear guidance as to how the colors, background, logo, typefaces can be customised.
To succeed in the second we need to be sure that there is a good range of contributed modules ported to Backdrop and, where there are problems with porting, we contribute team effort into getting this right.

**Expertise**
We are a small team and although this has merits, it means that the individuals with expertise are very busy and rarely have the time to spare to help novices or learners such as myself to understand some aspects of Backdrop. For example, I cannot yet deal with porting of entities and when discussion turns to ‘entity reference module’ my mind goes blank! Tutorials, worked examples, articles giving training at intermediate level seem essential if we are to grow the team. Another topic where I feel I need help to learn is the provision of tests within modules.

**Issue queue**
This seems to me to be excellent, though I fear that very useful reference information is getting hidden in the large volume of issues and comments. In order to highlight re-usable solutions could there be an extra flag ‘type – advice’ or similar?
Although each contributed module has its own issue queue, would it not help to have one common issue queue for flagging up, in particular, when help is needed?  Perhaps backdrop-ops/contrib is the place for this, in which case I would like to see a flag ‘help needed’ and a check on these at the weekly meetings. I say this in the belief that contributed modules are an essential part of Backdrop success.

**Forum**
I am very keen to see this happen. Is it a one person job or a team effort? If the latter, how can I participate? I hope there will be a section for recording ‘tips and tricks’ as perhaps a stepping-stone to the handbook.

**Weekly meetings**
Although I have never participated, largely because of time difference, I value being able to watch the videos which I try to do every week, listening and learning from what is said. I think it is a very important part of feeling to be in a team, so thank you for the time and effort. Maybe 30 minutes will prove adequate but please continue to make us feel we know what the issues are, and can understand any decisions made.

**Comment by serundeputy** [comment]
> Forum
I am very keen to see this happen. Is it a one person job or a team effort? If the latter, how can I participate? I hope there will be a section for recording ‘tips and tricks’ as perhaps a stepping-stone to the handbook.

@Graham-72 have a look at this and:

  1. If you agree with the direction say so.

  2.  Feel free to look at any of the issues in that queue and see if they seem like you could lend a hand or open new issues for direction feature request type things.

**Comment by jenlampton** [comment]
It would be great to review at the qa_forum as far as gathering ideas for what we might like on backdrop.cms.org, but @serundeputy that module is not a viable approach architecturally. The biggest problem is that answers/replies are not core's comments, meaning all the existing functionality we have for comments in core (and in contrib) - both in Backdrop and ported Drupal modules - would not work in our forum without additional integration. That's going to be too much work for those who maintain backdropcms.org.

But more importantly, it's probably not a good idea to talk about specific technical solutions before we know what we are building! :) We need a list of MVP features for launch, as well as a list of features we'd like to add later on. We are getting a pretty good list of features going in this issue, but there is no organization yet as to whether we will need things for launch, if we want to wait until they are needed, or if we even want them at all.

I have added a forum discussion item to our weekly meeting agenda next week, along with an agenda item to "talk about this more after the meeting" to help flush out some of these details.

I think it would be good to have a forum committee to help decide, build, and test the forum as it is in development. I was thinking @Graham-72 @serundeputy and myself (and maybe also @klonos?) for an initial team. Anyone else who is interested (or any of you who are not!) please let me know :)

@Graham-72 if after the weekly meeting doesn't work for you to chat next week, should we pick another time?  I would love it if you could join us in person. I can be available any time on Thursday (except during our other meetings, of course).

**Comment by serundeputy** [comment]
@docwilmot i nominate as chair.

**Comment by jenlampton** [comment]
> We are getting a pretty good list of features going, but there is no organization yet as to whether we will need things for launch, if we want to wait until they are needed, or if we even want them at all.

I just reread the whole thread and added all the recommended features from the comments into the list in the main post (guessed at organization). Let's use this as our starting point for the meeting next week, and see if we can make a decision on technical direction!

**Comment by Graham-72** [comment]
>@Graham-72 have a look at https://github.com/serundeputy/qa_forum and:

    If you agree with the direction say so.

    Feel free to look at any of the issues in that queue and see if they seem like you could lend a hand or open new issues for direction feature request type things.

@serundeputy I have had a quick look at what you have done and installed it on a test site. I even started trying building a couple of views but I'm afraid I get very distracted by poor styling of what the user sees and we desparately need the forthcoming new styling for comments in Basis. I will now wait for BD 1.6, the new styles and the meeting next Thursday before doing any more. Thanks for pointing me to what you have done.

**Comment by Graham-72** [comment]
>I have added a forum discussion item to our weekly meeting agenda next week, along with an agenda item to "talk about this more after the meeting" to help flush out some of these details.

@jenlampton yes, I am planning to join you for this week's meeting and the 'after the meeting' session. Is there anything I need to know about the technology for joining these?

**Comment by jenlampton** [comment]
I don't think so, you should be able to click the link for the google hangout, and it should "just work".  Famous last words! :)

**Comment by tomgrandy** [comment]
I have to use Firefox on my Mac for some reason and allow certain permissions to camera, etc.

Chrome, of all things, tends to hang on me.

**Comment by Graham-72** [comment]
@jenlampton @tomgrandy thanks for the advice. I have also found these reassuring pages:

For joining a hangout in progess:
https://support.google.com/hangouts/forum/AAAAfpFnBZ8uXmp537WmHI?hl=en

For muting microphone:
https://support.google.com/hangouts/answer/1254313?hl=en

I guess it's easy when you know how!

**Comment by jenlampton** [comment]
@docwilmot I'm wondering if you have a "Suggestions for 2018" post coming soon? How'd we do on the 2017 list?  Other than WE GOT A FORUM!!!!  (well, thanks - mostly - to you!)

**Comment by docwilmot** [comment]
Didnt plan to, but the issues in the list mostly stand, though we've improved on most.

**Comment by docwilmot** [comment]
@jenlampton just had a thought: can we have a midterm review session every cycle? So instead of the mad fury of reviews and merging that you and @quicksketch do on the eve of a major release, we have a slightly less furious version halfway as well.

**Comment by jenlampton** [comment]
@docwilmot that's a good idea, but how would we encourage such a thing? 

Maybe we could have an "official" sprint for 2 days, on release date + 2months, or something. So on March 14/15th, July 14/15th, and November 14/15th? We could encourage anyone who was interested in core to jump in on those two days and get their feet wet - it might also be a great time for mentoring and helping people get involved!

**Comment by docwilmot** [comment]
@jenlampton, yes that would be great if you and @quicksketch could manage the time. I think the idea of a 'deadline' does energize me to pick up old projects and PRs to try and get them over the finish line. But if you miss the end of cycle sprint you have to wait another 3 months, so that's actually a bit discouraging. I think this might help. 
Wonder what others think? @klonos @serundeputy

**Comment by serundeputy** [comment]
I like the idea.

**Comment by docwilmot** [comment]
2017 is well and truly past. Lots of good ideas, but no muscle to implement them. Closing.
